collector.go

```diff
@@ -50,10 +56,64 @@ func NewCollector(config CollectorConfig) (*Collector, error) {
 }
 
 func (c *Collector) Bundles() []Bundle {
-	return c.bundles
+	c.mutex.Lock()
+	defer c.mutex.Unlock()
+
+	return CopyBundles(c.bundles)
+}
+
+type CollectorEndpointResponse struct {
+	VersionBundles []Bundle `json:"version_bundles"`
 }
 
-// TODO
-func (c *Collector) Collect() error {
+func (c *Collector) Collect(ctx context.Context) error {
+	var responses [][]byte
+	{
+		var g errgroup.Group
```

Comment:
I would have done this in a for loop as an initial version, but this is cool too :D